packages/@react-aria/searchfield/docs/useSearchField.mdx

```diff
@@ -0,0 +1,163 @@
+style consistently cross browser. `useSearchField` helps achieve accessible 
+search fields that can be styled as needed.
+
+* Built with a native `<input type="search">` element
+* Visual and ARIA labeling support
+* Keyboard submit handling via the Enter key
+* Keyboard support for clearing the search field with the Escape key
+* Custom clear button support with internationalized label for accessibility
+
+## Anatomy
+
+Search fields consist of an input element, a label, and an optional clear button. 
+`useSearchField` automatically manages the labeling and relationships between the elements,
+and handles keyboard events. Users can press the Escape key to clear the search field, or the
+Enter key to trigger the `onSubmit` event.
+
+`useSearchField` returns two sets of props that you should spread onto the appropriate element:
+
+<TypeContext.Provider value={docs.links}>
+  <InterfaceType properties={docs.links[docs.exports.useSearchField.return.id].properties} />
+</TypeContext.Provider>
+
+State is managed by the `useSearchFieldState` hook in `@react-stately/searchfield`. The state object
+should be passed as an option to `useSearchField`.
+
+If there is no visual label, an `aria-label` or `aria-labelledby` prop must be passed instead 
+to identify the element to screen readers.
+
+## Example
+
+```tsx example
+import {useSearchFieldState} from '@react-stately/searchfield';
+
+function SearchField(props) {
+  let {label} = props;
+  let state = useSearchFieldState(props);
+  let ref = React.useRef();
+  let {labelProps, inputProps} = useSearchField(props, state, ref);
+
+  return (
+    <div style={{display: 'flex', flexDirection: 'column', width: 200}}>
+      <label {...labelProps}>{label}</label>
+      <input {...inputProps} ref={ref} />
+    </div>
+  );
+}
+
+<SearchField 
+  label="Search"
+  placeholder="What are you looking for?"
+  onSubmit={text => alert(text)} />
+```
+
+## Styling
+
+This example uses CSS to reset the default browser styling for a search field and implement
+custom styles. It also includes a custom clear button, using props from `useSearchField` and
+`usePress` (from @react-aria/interactions).
+
+```tsx example
+import {usePress} from '@react-aria/interactions';
+
+function SearchField(props) {
+  let {label} = props;
+  let state = useSearchFieldState(props);
+  let ref = React.useRef();
+  let {labelProps, inputProps, clearButtonProps} = useSearchField(props, state, ref);
+  let {pressProps} = usePress(clearButtonProps);
+
+  return (
+    <div className="search-field">
+      <label {...labelProps}>{label}</label>
+      <div>
+        <input {...inputProps} ref={ref} />
+        {state.value !== '' &&
+          <button {...clearButtonProps} {...pressProps}>❎</button>
+        }
+      </div>
+    </div>
+  );
+}
+
+<SearchField 
+  label="Search"
+  placeholder="What are you looking for?"
+  onSubmit={text => alert(text)} />
+```
+
+```css
```

Comment:
I assume you meant for this to get rendered out in the example as well?

Reply:
How do you mean? I think it does render...

Reply:
yes, it was just strange that we had css in the doc, but i can't think of a better way to do that